infraohjelmointi_api/serializers/TalpaProjectOpeningSerializer.py

```diff
@@ -0,0 +1,196 @@
+                  "majorDistrict", "majorDistrictName", "area", "unit", "isActive"]
+
+
+class TalpaProjectOpeningSerializer(serializers.ModelSerializer):
+    """
+    Serializer for TalpaProjectOpening with lock checking.
+    
+    Field Aliases (UI compatibility):
+    - projectStart -> projectStartDate
+    - projectEnd -> projectEndDate
+    - assetClassesId -> assetClassId (UI uses plural)
+    
+    Deprecated Fields (accepted but not used in Excel export):
+    - projectDescription
+    - responsiblePersonEmail
+    - responsiblePersonPhone
+    """
+    
+    # =========================================================================
+    # Foreign Key Fields - Read with nested serializer, write with ID
+    # =========================================================================
+    projectType = TalpaProjectTypeSerializer(read_only=True)
+    projectTypeId = serializers.PrimaryKeyRelatedField(
+        queryset=TalpaProjectType.objects.all(), source="projectType", write_only=True, required=False, allow_null=True
+    )
+    serviceClass = TalpaServiceClassSerializer(read_only=True)
+    serviceClassId = serializers.PrimaryKeyRelatedField(
+        queryset=TalpaServiceClass.objects.all(), source="serviceClass", write_only=True, required=False, allow_null=True
+    )
+    assetClass = TalpaAssetClassSerializer(read_only=True)
+    assetClassId = serializers.PrimaryKeyRelatedField(
+        queryset=TalpaAssetClass.objects.all(), source="assetClass", write_only=True, required=False, allow_null=True
+    )
+    # UI uses plural "assetClassesId" - accept both
+    assetClassesId = serializers.PrimaryKeyRelatedField(
+        queryset=TalpaAssetClass.objects.all(), source="assetClass", write_only=True, required=False, allow_null=True
+    )
+    # Project number range - replaces individual projectNumber
+    projectNumberRange = TalpaProjectNumberRangeSerializer(read_only=True)
+    projectNumberRangeId = serializers.PrimaryKeyRelatedField(
+        queryset=TalpaProjectNumberRange.objects.all(), source="projectNumberRange", write_only=True, required=False, allow_null=True
+    )
+    
+    # =========================================================================
+    # Metadata Fields
+    # =========================================================================
+    createdBy = PersonSerializer(read_only=True)
+    updatedBy = PersonSerializer(read_only=True)
+    isLocked = serializers.SerializerMethodField()
+    
+    # =========================================================================
+    # UI Field Aliases - Accept UI field names, map to model fields
+    # These allow the UI to use its preferred field names while the API
+    # uses the model field names internally.
+    # =========================================================================
+    # Schedule field aliases (UI uses projectStart/projectEnd, model uses projectStartDate/projectEndDate)
+    projectStart = serializers.DateField(
```

Comment:
Changed these projectStart and end in the UI also, so these aliases could be removed.